Blish HUD/Controls/ColorPicker.cs

```diff
@@ -10,106 +10,120 @@ namespace Blish_HUD.Controls {
+
+            colorBoxes = new Dictionary<Gw2Sharp.WebApi.V2.Models.Color, ColorBox>();
+        }
+
+        private void Colors_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) {
+            // Remove any items that were removed (first because moving an item in a collection will put
+            // that item in both OldItems and NewItems)
+            if (e.OldItems != null) {
+                foreach (var deletedItem in e
+                                       .OldItems
+                                       .Cast<Gw2Sharp.WebApi.V2.Models.Color>()
+                                       .Where(delItem => colorBoxes.ContainsKey(delItem))) {
+                    colorBoxes[deletedItem].Dispose();
+                    colorBoxes.Remove(deletedItem);
+                }
+            }
+
+            if (e.NewItems != null) {
+                foreach (Gw2Sharp.WebApi.V2.Models.Color addedItem in e.NewItems) {
+                    if (!colorBoxes.ContainsKey(addedItem)) {
+                        var colorBox = new ColorBox() {
+                            Color = addedItem,
+                            Parent = this,
+                            Size = new Point(COLOR_SIZE),
+                        };
+
+                        colorBoxes[addedItem] = colorBox;
+
+                        colorBox.LeftMouseButtonPressed += delegate {
+                            foreach (var box in colorBoxes.Values) {
+                                box.Selected = false;
+                            }
+
+                            colorBox.Selected = true;
+                            this.SelectedColor = colorBox.Color;
+                        };
+                    }
+                }
+            }
+
+            // Relayout the color grid
+            for (int i = 0; i < this.Colors.Count; i++) {
+                var currentColor = this.Colors[i];
+                var currentBox = colorBoxes[currentColor];
+
+                int horizontalPosition = i % colorsPerRow;
+                int verticalPosition = i / colorsPerRow;
+
+                currentBox.Location = new Point(
+                    horizontalPosition * (currentBox.Width + COLOR_PADDING),
+                    verticalPosition * (currentBox.Width + COLOR_PADDING)
+                );
+            }
+        }
+
+        public override void Invalidate() {
+            base.Invalidate();
+
+            colorsPerRow = (this.Width -10 ) / (COLOR_SIZE + COLOR_PADDING);
+            this.ContentRegion = new Rectangle(COLOR_PADDING, COLOR_PADDING, (this.Width - 10) - (COLOR_PADDING * 2), this.Height - (COLOR_PADDING * 2));
+        }
+
+        public override void PaintBeforeChildren(SpriteBatch spriteBatch, Rectangle bounds) {
+            // Draw background
+            spriteBatch.DrawOnCtrl(this, Content.GetTexture(BACKGROUND_TEXTURE_NAME), bounds, Color.Black * 0.5f);
```

Comment:
The content service does cache the textures, but we no longer load textures in the draw loop where we can help it.  I'd assign it somewhere statically as we did with the `ColorBox` control and reference that.  This was just still around from my old code before we made the transition to non-cached UI renders and it didn't matter.